spec.md

```diff
@@ -148,20 +148,47 @@ specific protocols (AWS Kinesis, Azure Event Grid).
 
 ## Type System
 
-The following abstract data types are available for use in attributes.
+The following abstract data types are available for use in attributes. Each of
+these types MAY be represented differently by different event formats and in
+transport metadata fields. This specification defines a canonical string-encoding
+for each type that MUST be supported by all implementations.
+
+A strongly-typed programming model that represents a CloudEvent or any 
+extension MUST be able to convert from and to the canonical string-encoding to
+the runtime/language native type that best corresponds to the abstract type.
+
+For example, the `time` attribute might be represented by the language's native
+*datetime* type in a given implementation, but it MUST be settable providing
```

Comment:
"MUST be settable"... not sure what this means. Does this mean the SDK's setter MUST accept a "string" value from the user?  Or is this more about how there are certain transports that will use the string-encoding and impls that support that transport must be prepared to convert between strings and the native type?

Reply:
Oh, the next sentence covers the transport stuff. So I think I'm back to not understanding what the "settable" part means in practice.

Reply:
@duglin "settable" means that the implementation MUST ALWAYS allow the app to set a `time` value via its string representation, either from the SDK side (e.g. you're building a forwarder that maps between transports) or from the wire side, even if the "native" representation of the type isn't string. E.g.  set_time(time_t) and set_time(string).

Reply:
ok that helps clarify things- thanks.

However, is this out of scope for the spec? Should this be guidance in the sdk doc? I'm not sure the spec, which is just about (mainly) defining the metadata, should get into how the metadata is set/retrieved by the app/user.  Especially via a "MUST" - as non-normative guidance I can see though.